editor/editor_export.cpp

```diff
@@ -151,7 +151,7 @@ void EditorExportPreset::set_export_path(const String &p_path) {
 	/* NOTE(SonerSound): if there is a need to implement a PropertyHint that specifically indicates a relative path,
 	 * this should be removed. */
 	if (export_path.is_abs_path()) {
-		String res_path = OS::get_singleton()->get_resource_dir();
+		String res_path = OS::get_singleton()->get_resource_dir()+"/";
```

Comment:
Style issue, there should be spaces around `+`.

Is the extra `/` always needed, or only if `get_resource_dir()` somehow lacks it?
If it's the latter, you can use `plus_file("")`, which is a hacky way to ensure that `/` is appended only if it's missing (so we don't end up with `some/path//`).

Reply:
Alternatively, `path_to_file()` could maybe be fixed to properly add a directory separator if the String it operates on lacks it?

Reply:
It looks like when the value of `get_resource_dir()` is initialized the slash on the end is consistently removed (see line 460 of project_settings.cpp).

For sure, `path_to_file()` could be modified to include `plus_file("")`. That sounds better if `get_resource_dir()` is one of many directory paths lacking a `/` at the end. I'm not sure though.

Reply:
> For sure, `path_to_file()` could be modified to properly add a separator. That sounds better if `get_resource_dir()` is one of many directory paths lacking a separator at the end. I'm not sure if that's the case, though.

`path_to_file` is not part of the public scripting API for String, but it's exposed in GDNative, so in theory GDNative plugins can call it, and might do it with non slash-terminated paths.

I would expect `"/home/akien".path_to_file("bin/godot")` to give me `/home/akien/bin/godot` (otherwise I might just as well use `+`), so IMO it's best to fix this one instead of fixing all calling sites.

Reply:
Something like this might work:
```diff
diff --git a/core/ustring.cpp b/core/ustring.cpp
index 6a1dc2295f..345db0eb13 100644
--- a/core/ustring.cpp
+++ b/core/ustring.cpp
@@ -3792,7 +3792,8 @@ bool String::is_valid_float() const {
 
 String String::path_to_file(const String &p_path) const {
 
-	String src = this->replace("\\", "/").get_base_dir();
+	// Don't get base dir for src, this is expected to be a dir already.
+	String src = this->replace("\\", "/");
 	String dst = p_path.replace("\\", "/").get_base_dir();
 	String rel = src.path_to(dst);
 	if (rel == dst) // failed
diff --git a/editor/editor_file_dialog.cpp b/editor/editor_file_dialog.cpp
index 7fc989b5e9..7bf7cd17ec 100644
--- a/editor/editor_file_dialog.cpp
+++ b/editor/editor_file_dialog.cpp
@@ -958,6 +958,9 @@ String EditorFileDialog::get_current_path() const {
 }
 void EditorFileDialog::set_current_dir(const String &p_dir) {
 
+	if (p_dir.is_rel_path()) {
+		dir_access->change_dir(OS::get_singleton()->get_resource_dir());
+	}
 	dir_access->change_dir(p_dir);
 	update_dir();
 	invalidate();
```

Reply:
I can confirm that the above patch fixes both #31840 and #31427.
Would you mind amending the commit to do those changes?

Reply:
Awesome, fixed!